modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/display/context/AssetCategoriesNavigationDisplayContext.java

```diff
@@ -190,6 +201,62 @@ public long getDisplayStyleGroupId() {
 		return _displayStyleGroupId;
 	}
 
+	private String[] _getAssetVocabularyIds() {
+		PortletPreferences portletPreferences = _renderRequest.getPreferences();
+
+		Map<String, String[]> map = portletPreferences.getMap();
+
+		if (!map.containsKey("assetVocabularyGroupExternalReferenceCodes")) {
+			return _assetCategoriesNavigationPortletInstanceConfiguration.
+				assetVocabularyIds();
+		}
+
+		List<Long> assetVocabularyIds = new ArrayList<>();
+
+		String[] assetVocabularyGroupExternalReferenceCodes =
+			GetterUtil.getStringValues(
+				portletPreferences.getValues(
+					"assetVocabularyGroupExternalReferenceCodes", null));
+
+		for (String assetVocabularyGroupExternalReferenceCode :
+				assetVocabularyGroupExternalReferenceCodes) {
+
+			String[] assetVocabularyExternalReferenceCodes =
+				GetterUtil.getStringValues(
+					portletPreferences.getValues(
+						"assetVocabularyExternalReferenceCodes_" +
+							assetVocabularyGroupExternalReferenceCode,
+						null));
```

Comment:
Shouldn't this be moved after the group check? If the group is null, this value won't be used. There's an SF rule that checks these kinds of things; it may not have caught this particular case, but this is a general pattern.

Reply:
You are right, thanks @adolfopa!